### PR TITLE
Feature/ret 1631 implement filter for diff command

### PR DESCRIPTION
**Pull request by ivonx**

### CODE REVIEW

File: src/test/java/de/retest/recheck/ignore/SearchFilterFilesTest.java [hunk 1]
```diff
@@ -62,4 +67,20 @@ void getProjectFilterFiles_should_only_get_filter_files() throws IOException {
 		assertThat( projectFilterFiles.stream().map( Path::getFileName ) ).contains( colorFilter.toPath().getFileName(),
 				webFontFilter.toPath().getFileName() );
 	}
+
+	@Test
+	void searchFilterByName_should_return_filter_file() throws IOException {
+		final String name = "positioning.filter";
+		final String invalidName = "color.filter";
```
Comment (roesslerj) [inline src/test/java/de/retest/recheck/ignore/SearchFilterFilesTest.java]: I would split that into two tests ... this one and searchFilterByName_should_not_return_nonexistent_file mit color.filter.

**Review by roesslerj — CHANGES_REQUESTED** [review]

File: src/main/java/de/retest/recheck/ignore/CompoundFilter.java [hunk 2]
```diff
@@ -13,6 +13,10 @@ public CompoundFilter( final List<Filter> filters ) {
 		this.filters = filters;
 	}
 
+	public List<Filter> getFilters() {
```
Comment (diba1013) [inline src/main/java/de/retest/recheck/ignore/CompoundFilter.java:17]: You just need this method in a test? I think, that you could just use the Filter methods to compare [this ](https://github.com/retest/recheck.cli/pull/62/files#diff-f4a532528223635b5a54a9dadb6877ceR86). If you really need this method, make it immutable `Collections.immutableList`, maybe even within the constructor.

File: src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java [hunk 3]
```diff
@@ -56,4 +58,21 @@ private SearchFilterFiles() {}
 			return Collections.emptyList();
 		}
 	}
+
+	public static Optional<Filter> searchFilterByName( final String name ) throws IOException {
+		for ( int j = 0; j < getAllFilterFiles().size(); j++ ) {
```
Comment (diba1013) [inline src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java]: You are calling `getAllFilterFiles()` multiple times. Please extract a variable.

Comment (diba1013) [inline src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java]: You do not need the index here. Please consider a `for-each-loop` or advanced, a stream 😄.

File: src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java [hunk 4]
```diff
@@ -56,4 +58,21 @@ private SearchFilterFiles() {}
...
+			final Path filterPath = getAllFilterFiles().get( j );
+			if ( filterPath.getFileName().toString().equals( name ) ) {
+				log.info( "Loading filter file from path: " + filterPath );
+				try ( final Stream<String> ignoreFileLines = Files.lines( filterPath ) ) {
+					final Filter ignoreApplier = Loaders.load( ignoreFileLines ) //
+							.filter( Filter.class::isInstance ) //
+							.map( Filter.class::cast ) //
+							.collect( Collectors.collectingAndThen( Collectors.toList(), CompoundFilter::new ) );
+					return Optional.of( ignoreApplier );
```
Comment (diba1013) [inline src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java:71]: We really need one method for loading filters someday...

**Review by diba1013 — CHANGES_REQUESTED** [review]

**Comment by rebazer** [comment]
Rebase failed, this pull request needs some manual love ...

**Comment by beatngu13** [comment]
Blocked by RET-1720, let's reopen afterwards.

**Comment by rebazer** [comment]
Rebase failed, this pull request needs some manual love ...

**Comment by rebazer** [comment]
Rebase failed, this pull request needs some manual love ...

**Comment by beatngu13** [comment]
Close until conflicts have been resolved.

**Pull request closed without merging**